### PR TITLE
added file share code to store triage ids

### PR DESCRIPTION
**Pull request by nisha-das-metron**
Required items, please complete
   
   Change(s):
   - Updated the function app code and ARM template
   - Added functionality to store triage id's in a file share
   - Added functionality to fetch data from API as well as from files and process only 150 data and update the file with remaining 
   data

   Reason for Change(s):
   - Change in the limitation of generating alerts in the execution of one analytical rule
   - Duplicate incidents getting created
   - Status is not getting updated timely

   Version Updated:
   - Yes
   - Updated version of the function app

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by nisha-das-metron** [comment]
> 



> @nisha-das-metron please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Digital Shadows Searchlight"

**Comment by v-prasadboke** [comment]
Hello @nisha-das-metron looking into this

**Comment by v-prasadboke** [comment]
Hello @nisha-das-metron working on this

**Comment by v-sabiraj** [comment]
Hey @nisha-das-metron, can you also please add the missing link for function app code in this file (https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Digital%20Shadows/Data%20Connectors/Digital%20Shadows/DigitalShadowsSearchlight_API_functionApp.json), Check the highlighted part of image. Thanks.

image

**Comment by nisha-das-metron** [comment]
Hi @v-sabiraj, I have added the GitHub link

File: Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsSearchlight_API_functionApp.json [hunk 1]
```diff
@@ -119,7 +119,7 @@
...
+            "description": "1. In the newly created Function App, select **Functions** from the navigation menu and click **+ Add**.\n2. Select **Timer Trigger**.\n3. Enter a unique Function **Name** in the New Function field and leave the default cron schedule of every 5 minutes, then click **Create Function**.\n4. Click on the function name and click **Code + Test** from the left pane.\n5. Copy the [Function App Code](https://github.com/digitalshadows/Azure-Sentinel/tree/digital-shadows-version_2.0.2/Solutions/Digital%20Shadows/Data%20Connectors/Digital%20Shadows/DigitalShadowsConnectorAzureFunction) and paste into the Function App `run.ps1` editor.\n6. Click **Save**."
```
Comment (v-sabiraj) [inline Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsSearchlight_API_functionApp.json:122]: Hey @nisha-das-metron, the link added is not redirecting to correct path, by looking to the link added I think the path is given to add "DigitalShadowsConnectorAzureFunction" so got this link with correct path, so can you please add this in PR? https://github.com/digitalshadows/Azure-Sentinel/tree/digital-shadows-version_2.0.2/Solutions/Digital%20Shadows/Data%20Connectors/Digital%20Shadows/DigitalShadowsConnectorAzureFunction)

**Comment by v-sabiraj** [comment]
Hey @nisha-das-metron, can you please check on the comment added, thanks.

**Comment by nisha-das-metron** [comment]
hey @v-sabiraj, can you  please check on the comment added.

Thanks

**Comment by nisha-das-metron** [comment]
@v-sabiraj  @v-prasadboke 
Please let us know if you require any further changes. Thanks

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj @nisha-das-metron has made some changes

**Comment by v-sabiraj** [comment]
> Hi, @v-sabiraj I have changed the steps for function app deployment in DigitalShadowsSearchlight_API_functionApp.json file. can you check once?
> 
> Also, I have modified the digitalshadowsConnector.zip file and pushed the modified file in the digital-shadows-version_2.0.2 branch and added the GitHub link for the zip file in DigitalShadowsSearchlight_API_functionApp.json file
> 
> Thanks

Hi @nisha-das-metron, can you please add screenshots of successful invocation logs, actually I don't have appropriate credentials to get the logs, thanks.

**Comment by nisha-das-metron** [comment]
Hi, @v-sabiraj @v-prasadboke  Please find attached the screenshot of the execution logs. Thanks

[execution_log-2.pdf](https://github.com/Azure/Azure-Sentinel/files/11517050/execution_log-2.pdf)
[execution_log-1.pdf](https://github.com/Azure/Azure-Sentinel/files/11517052/execution_log-1.pdf)

**Comment by alexandrupnt** [comment]
Hello,

Need to ask when the Azure SENTNEL Connector for digital shadow will be updated,? As of right now the link in the Connector are unavailable for the functions
image

the link for the code returns page not found

**Comment by v-sabiraj** [comment]
Hey @nisha-das-metron, thanks for the execution logs, we will get this PR merged, can you please update the solution package once we merge this pr, thanks.

**Comment by alexandrupnt** [comment]
Hello, any updates on this topic ?

**Comment by nisha-das-metron** [comment]
The new version has been released

**Pull request merged**